packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -321,11 +339,61 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
     assert(newDelegate != null);
     if (_delegate == newDelegate)
       return;
-    if (newDelegate.runtimeType != _delegate.runtimeType || newDelegate.shouldRelayout(_delegate))
+    if (newDelegate.runtimeType != _delegate.runtimeType || newDelegate.shouldRelayout(_delegate)) {
+      _specification = null;
       markNeedsLayout();
+    }
     _delegate = newDelegate;
   }
 
+  int get initialChildIndex => _initialChildIndex;
+  int _initialChildIndex;
+  void set initialChildIndex(int value) {
+    assert(value != null);
+    if (_initialChildIndex == value)
```

Comment:
Reading this, I wonder if we should write an update(slot, value, callIfChanged) helper if we're writing tons of these.  Not for this patch of course.